wsd/DocumentBroker.cpp

```diff
@@ -1491,9 +1491,33 @@ void DocumentBroker::handleUploadToStorageResponse(const StorageBase::UploadResu
             Poco::URI::encode(filename, "", encodedName);
             const std::string filenameAnonym = LOOLWSD::anonymizeUrl(filename);
 
+            constexpr const char* needle = "?access_token";
+            std::size_t index = getPublicUri().toString().find(needle);
```

Comment:
This is problematic because access_token doesn't have to be the first parameter, it could be later, in which case it will be "&access_token". Perhaps search for "access_token=" ? (But best to use Poco::URI and getQueryParameters).